src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -383,6 +383,26 @@ public object TrySteal(ref bool missedSteal)
                     return null;
                 }
             }
+
+            public int Count
+            {
+                get
+                {
+                    bool lockTaken = false;
+                    try
+                    {
+                        m_foreignLock.Enter(ref lockTaken);
```

Comment:
Do we need to lock, or could we just say that the count is an estimate and hot read the head and tail?

Reply:
I was thinking about that but in the overflow case of the tail index the hot reads may yield a totally wrong result if read during the update. I figured perf is not that big an issue so may as well do the safe thing.